src/main.py

```diff
@@ -646,8 +652,9 @@ def non_ha_replica_iter(self, db_state, zk_state):
                     wal_receiver_info and wal_receiver_info[0].get('status') == 'streaming'
                 )
                 logging.error(replication_source_replica_info)
-                current_primary = zk_state['lock_holder']
+
                 if replication_source_is_dead:
+                    self._acquire_replication_source_lock(current_primary)
```

Comment:
It's an extra, you are right. Because there is _acquire_replication_source_lock call in _return_to_cluster function (below this line).
But we should acquire lock:
- When we are streaming, because if not, the lock will not be held, but the slot still exists, so if we don't acquire the lock, the replication slot will be dropped by the source.
- When we aren't streaming. In the event of long replica unavailability or a changing replication source.